stoobly_agent/app/cli/helpers/endpoints_import_service.py

```diff
@@ -1,123 +1,153 @@
         'query': response_param_name.get('query'),
         'endpoint_id': endpoint_id,
-        'project_id': context.project_id,
+        'project_id': project_id,
     })
 
-    res.raise_for_status()
-
-    return res.json()
+    return res
 
-def import_response_header_name(context: EndpointsImportContext, endpoint_id: int, response_header_name: RequestComponentName):
-    res: requests.Response = context.response_header_name_resource.create(**{
+def import_response_header_name(project_id: int, endpoint_id: int, response_header_name_resource: ResponseHeaderNamesResource, response_header_name: RequestComponentName):
+    res: requests.Response = response_header_name_resource.create(**{
         'name': response_header_name.get('name'),
         'is_deterministic': response_header_name.get('is_deterministic', True),
         'is_required': response_header_name.get('is_required', False),
         'endpoint_id': endpoint_id,
-        'project_id': context.project_id,  
+        'project_id': project_id,  
     })
 
-    res.raise_for_status()
+    return res
+
+component_name_import_dispatch = {
+    'header_name': import_header_name,
+    'body_param_name': import_body_param_name,
+    'query_param_name': import_query_param_name,
+    'response_param_name': import_response_param_name,
+    'response_header_name': import_response_header_name
+}
+
+def process_import(args: PendingImportArgs):
+    return component_name_import_dispatch[args['component_name']](args['project_id'], args['endpoint_id'], args['component_resource_class'], args['component'])
+
+def import_endpoint_and_component_name(context: EndpointsImportContext, endpoint: EndpointShowResponse, uploaded_endpoints):
+    res = import_endpoint(context.project_id, context.resources['endpoint'], endpoint)
+
+    if not res.ok:
+        return res
+
+    endpoint_id = res.json()['id']
+    uploaded_endpoints.append(endpoint_id)
+    import_queue = []
+    for component_name in ENDPOINT_COMPONENT_NAMES:
+        for component in endpoint.get(f'{component_name}s', {}):
+            resource = context.resources[component_name]
+            pending_import: PendingImportArgs = {
+                'project_id': context.project_id,
+                'endpoint_id': endpoint_id,
+                'component_name': component_name,
+                'component': component,
+                'component_resource_class': resource
+            }
+            import_queue.append(pending_import)
+    
+    while res.ok and len(import_queue) > 0:
+        pending_import = import_queue.pop()
+        res = process_import(pending_import)
+    
+    return res
+
+def cleanup_endpoints(project_id: int, resource: EndpointsResource, uploaded_endpoints):
+    for endpoint_id in uploaded_endpoints:
+        res: requests.Response = resource.destroy(endpoint_id, **{
+            'project_id': project_id
+        })
 
-    return res.json()
\ No newline at end of file
+        res.raise_for_status()
```

Comment:
We can raise an error here, but we should catch it somewhere. Alternatively, we can log the endpoint so the user at least has the information to retry on their own at some later time.